command/agent/agent.go

```diff
@@ -127,28 +130,64 @@ func Create(config *Config, logOutput io.Writer) (*Agent, error) {
 		logOutput = os.Stderr
 	}
 
-	// Validate the config
-	if config.Datacenter == "" {
-		return nil, fmt.Errorf("Must configure a Datacenter")
-	}
-	if config.DataDir == "" && !config.DevMode {
-		return nil, fmt.Errorf("Must configure a DataDir")
-	}
-
-	// Try to get an advertise address
+	var bindIPs []string
```

Comment:
Why is this up at this scope but not used anywhere except the one clause? Did you mean to stick the advertise address into this for each of these?